mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs

```diff
@@ -90,8 +90,11 @@ internal TreeNodeCollection (TreeNode owner)
 			set {
 				if (index < 0 || index >= Count)
 					throw new ArgumentOutOfRangeException ("index");
-				SetupNode (value);
+				
+				OnUIACollectionChanged (CollectionChangeAction.Remove, nodes [index]);
 				nodes [index] = value;
```

Comment:
The order of operations is reversed here. While the UIA API is not well documented this seems like a breaking change. 

```csharp
var removed = nodes [index];
nodes [index] = value;
OnUIACollectionChanged (CollectionChangeAction.Remove, removed);
```

Reply:
@filipnavara, thank you! You are right, one should raise "remove"- event after the node is removed. How do you like this approach:
```
    set {
        if (index < 0 || index >= Count)
            throw new ArgumentOutOfRangeException ("index");
        
-       OnUIACollectionChanged (CollectionChangeAction.Remove, nodes [index]);
-       nodes [index] = value;
-       
-       SetupNode (value);
+       RemoveAt (index, false);
+       Insert (index, value);
    }
```

Reply:
That will cause unnecessary array resizing which could be costly. I'd go with the less disruptive change.

Reply:
Please, find the PR update